### PR TITLE
miscellaneous fixes to shooter and climber

### PR DESCRIPTION
**Pull request by thetaspirit**
* the servo that releases the climber now stays at its locked position when robot is disabled, then re-enabled
* using beam break sensor for shooter
* making sure shooter rpm has been up to speed for a certain amount of ticks
* tuned vision pid
* added minor friction coefficients for drivetrain, needs tuning but doesn't cause issues right now

### CODE REVIEW

File: code/src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -240,6 +250,7 @@ public void teleopInit() {
   @Override
   public void teleopPeriodic() {
     _robot.teleopPeriodic();
+    SmartDashboard.putNumber("pdp Voltage", _pdp.getVoltage());
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/Robot.java]: remove all references to the pdp in robot, not needed on master

File: code/src/main/java/frc/robot/RobotConfig.java [hunk 2]
```diff
@@ -38,6 +38,9 @@
         public final int intakeInButtonPort = XboxController.Button.kBumperRight.value;
         public final int intakeOutButtonPort = XboxController.Button.kBumperLeft.value;
 
+        public final int shooterIncreaseButton = 0;
+        public final int shooterDecreaseButton = 180;
+
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/RobotConfig.java:43]: changes these variable names

File: code/src/main/java/frc/robot/RobotConfig.java [hunk 3]
```diff
@@ -126,6 +131,7 @@
         public final int leftShooterPort = 16;
         public final int rightShooterPort = 17;
 
+        public final double lowerBangValue = .9;
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/RobotConfig.java]: change this name, or write a comment

File: code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 4]
```diff
@@ -32,23 +34,27 @@
...
+    private RobotConfig _config;
     private boolean m_hasShotBall = true; // ONE ball, singular!
     private boolean m_isShooting = false;
 
     private final int pidOnTargetTicks; // the number of ticks the pid must be on target for to be considered ready to
                                         // shoot
     private int m_onTargetFor = 0; // the amount of ticks the pid has been on target
 
-    private final int shooterRPM; //
+    private int targetShooterRPM;
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: change name

File: code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 5]
```diff
@@ -62,27 +68,24 @@ public ShooterSubsystem(ILogger logger, final RobotConfig config) {
...
-        shooterRPM = config.Shooter.shooterRPM;
+        this.targetShooterRPM = config.Shooter.shooterRPM;
 
         m_kicker = new WPI_VictorSPX(config.Shooter.feederPort);
         m_wheels = new WPI_VictorSPX(config.Shooter.spinnerPort);
         m_wheels.setInverted(true);
 
         m_feeder = new SpeedControllerGroup(m_kicker, m_wheels);
 
+        m_beambreak = new TimeOfFlight(51);
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: move this to configs

File: code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 6]
```diff
@@ -112,43 +115,49 @@ public boolean getHasShotBall() {
...
-        double shooterSpeed = m_shooterPID.calculate(m_leftEncoder.getVelocity());
-        if (shooterSpeed > -0.2)
-            m_shooter.set(shooterSpeed);
-        else {
-            m_shooter.set(-0.2);
-        }
-        if (m_shooterPID.getSetpoint() == 0) {
+        /*
+         * double shooterSpeed = m_shooterPID.calculate(m_leftEncoder.getVelocity()); if
+         * (shooterSpeed > -0.2) m_shooter.set(shooterSpeed); else {
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: delete code commented out

File: code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 7]
```diff
@@ -112,43 +115,49 @@ public boolean getHasShotBall() {
...
                 this.m_isShooting = true;
             }
         }
 
-        // if we are shooting and the pid isn't at the setpoint, that means we've shot a
-        // ball
-        if (this.m_isShooting && !m_shooterPID.atSetpoint()) {
+        // if we are/were shooting and the pid isn't at the setpoint, that means we've
+        // shot a ball
+        if (this.m_isShooting && this.hasBeamBroken() && m_onTargetFor > 5) {
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: move this to configs

File: code/src/main/java/frc/robot/subsystems/SwerveDrivetrain.java [hunk 8]
```diff
@@ -149,9 +149,19 @@ public void move(double x, double y, double theta, final boolean fieldRelative)
 
                 }
                 // applies heading correction only when moving and not sending a input command
-                else if (x != 0 && y != 0) {
+                else if (x != 0 || y != 0) {
                         theta = this.m_pidController.calculate((((this.m_gyro.getAngle() % 360) + 360) % 360));
                 }
+
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/subsystems/SwerveDrivetrain.java:155]: comment on this

File: code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 7 above)
Comment (Meghd2cx) [reply code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: renamed to m_minOnTargetTicks

Justification: This is the minimum number of ticks that the limelight must be on target for it to be considered "ready to shoot".

File: code/src/main/java/frc/robot/Robot.java [hunk 9]
```diff
@@ -117,7 +117,14 @@ private void configureButtonBindings() {
     winchDirectionControl.whenReleased(_container.getInstance(SetClimbNormal.class));
 
     JoystickButton start = new JoystickButton(_operatorController, _config.Controller.shooterStartupButtonPort);
-    start.whenPressed(_container.getInstance(ShooterStartupCommand.class));
+    start.whenPressed(_container.getInstance(ShooterStartupCommand.class).withRPM(_config.Shooter.shooterRPM));
+
```
Comment (thetaspirit) [inline code/src/main/java/frc/robot/Robot.java:121]: give both the names of the shooter buttons and rpm variables better, more descriptive names.  use the location of the field (trench, sector, etc) in the name

**Review by Meghd2cx — APPROVED** [review]

**Pull request merged**